tests/utils/pod.go

```diff
@@ -84,12 +84,13 @@ func NewPodWithPVC(podName, cmd string, pvc *k8sv1.PersistentVolumeClaim) *k8sv1
 // Finds the first pod which has the passed in prefix. Returns error if multiple pods with the same prefix are found.
 func FindPodByPrefix(clientSet *kubernetes.Clientset, namespace, prefix, labelSelector string) (*k8sv1.Pod, error) {
 	var result k8sv1.Pod
-	var multiplePods bool
+	var multiplePods, foundPod bool
 	err := wait.PollImmediate(2*time.Second, PodCreateTime, func() (bool, error) {
 		multiplePods = false
-		podList, err := clientSet.CoreV1().Pods(namespace).List(metav1.ListOptions{})
+		podList, err := clientSet.CoreV1().Pods(namespace).List(metav1.ListOptions{
```

Comment:
Line above: `multiplePods = false`
You don't need to do this. False is its default/"zero" value. Once it becomes true you want to exit the Poll. IOW, it doesn't matter if you have 2 pods w/ same prefix or 100, once there's >1 exit the Poll.

Reply:
Actually, I don't think you need `multiplePods`, see below

Reply:
I am missing something, I need multiplePods to let function outside the poll loop know I found multiples. In each poll loop I loop over all found pods and thus I can find multiples.

Reply:
That happens with the refactor because you're inner func is returning _true_ to stop the Poll and _err_ which is then returned by `PollImmediate`

Reply:
Yes I got it now, tricky little details.